### PR TITLE
Feature: reboot service

### PR DESCRIPTION
**Pull request by EdLeckert**
Create new service to reboot OpenSprinkler controller.

### CODE REVIEW

**Comment by vinteo** [comment]
Thanks for this!!

Is it possible to add an example to the README as well?

File: custom_components/opensprinkler/__init__.py [hunk 1]
```diff
@@ -26,9 +26,11 @@
     SCHEMA_SERVICE_RUN,
     SCHEMA_SERVICE_SET_WATER_LEVEL,
     SCHEMA_SERVICE_STOP,
+    SCHEMA_SERVICE_REBOOT,
```
Comment (vinteo) [inline custom_components/opensprinkler/__init__.py:29]: these should be in alphabetical order

File: custom_components/opensprinkler/const.py [hunk 2]
```diff
@@ -32,6 +32,9 @@
     vol.Required(CONF_WATER_LEVEL): vol.All(vol.Coerce(int), vol.Range(min=0, max=250)),
 }
 
+SCHEMA_SERVICE_REBOOT = {}
+
 SERVICE_RUN = "run"
 SERVICE_STOP = "stop"
 SERVICE_SET_WATER_LEVEL = "set_water_level"
+SERVICE_REBOOT = "reboot"
```
Comment (vinteo) [inline custom_components/opensprinkler/const.py:40]: Add extra line at the end

**Comment by vinteo** [comment]
Just some minor linting stuff and an example for the README.

**Comment by EdLeckert** [comment]
Hey, Vincent. Thanks for the feedback. 

 

This is a trial Pull Request for me…my first ever, so just figuring out the process. I was a career software dev, but haven’t used git in almost 10 years, and never on a public project…only internal.

 

So, I don’t have tests working locally and never played with linting, for example, but I’ll figure it out.

 

The good news, at least for me, is that I have a lot of other features already working on my HA instance, such as updating program station durations, start/interval days, and some others. If I can get the process figured out, I’d be happy to submit those as well.

 

Regards,

Ed

**Comment by vinteo** [comment]
No worries, take your time and if there are any questions just ask.

Also the other thing I would recommend is to squash all the commits to one when you are done

**Comment by EdLeckert** [comment]
OK, I think my changes are ready.

 

What’s your recommended safest way to squash my entire branch, and what’s the process to get the changes to you? A new pull request?

 

Thanks,

Ed

**Comment by EdLeckert** [comment]
UPDATE:

 

So I see that the changes are automatically added to the pull request when I commit them to my branch. Cool.

 

I’m still confused as to the best way to get the changes squashed. I certainly agree that it needs to happen, but reading the docs, it looks to me like you can do it automatically on the merge. Is that not so?

 

From https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/incorporating-changes-from-a-pull-request/about-pull-request-merges#squash-and-merge-your-commits :

 <https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/incorporating-changes-from-a-pull-request/about-pull-request-merges#squash-and-merge-your-commits> Squash and merge your commits

When you select the Squash and merge option on a pull request on GitHub.com, the pull request's commits are squashed into a single commit. Instead of seeing all of a contributor's individual commits from a topic branch, the commits are combined into one commit and merged into the default branch. Pull requests with squashed commits are merged using the fast-forward option <https://git-scm.com/docs/git-merge#_fast_forward_merge> .

They’re describing the process the project owner does, right?

 

Once again, I’m completely new to using github on open projects, so I appreciate your patience!

 

******************

OK, I think my changes are ready.

 

What’s your recommended safest way to squash my entire branch, and what’s the process to get the changes to you? A new pull request?

 

Thanks,

Ed

**Comment by vinteo** [comment]
Some instructions: https://www.freecodecamp.org/news/git-squash-commits/

Then do a force push `git push -f origin [branch name]`

The PR will update automatically

Yes it is possible to do it during merge if you rather not do it but I feel manually squashing is cleaner and you have more control over the commit message.

**Comment by EdLeckert** [comment]
So I did all this and got a successful linting, but it doesn’t appear that anything has happened on your side to rerun the actions. Do you need to do something or do I?

**Comment by vinteo** [comment]
You haven't pushed your changes to github: `git push -f origin feat-reboot-service`

**Comment by EdLeckert** [comment]
Closing to reopen with renamed branch.

**Comment by EdLeckert** [comment]
“You haven't pushed your changes to github: git push -f origin feat-reboot-service”

That wasn’t it. As I said, I followed your instructions and observed the changes in github, including a successful linting action.

The actual problem was that my branch name got renamed at some point and the link to the PR got broken.

Seemed easier to close the original PR (with its comments irrelevant to the actual feature) and start over. Hope that doesn’t cause you any issues.

**Pull request closed without merging**